frontend/src/components/mine/Tailings/MineTailingsInfo.js

```diff
@@ -77,11 +78,16 @@ export class MineTailingsInfo extends Component {
   };
 
   handleAddReportSubmit = (value) => {
-    const requiredReportLabel = this.getMineTSFRequiredDocumentsHash[value.req_document_guid];
-    this.createMineExpectedDocument(this.props.mine.guid, {
-      document_name: requiredReportLabel,
-      ...value,
-    }).then(() => {
+    const requiredReport = this.props.mineTSFRequiredReports.find(
+      (x) => x.req_document_guid === value.req_document_guid
+    );
+    const newRequiredReport = {
+      document_name: requiredReport.req_document_name,
+      req_document_guid: requiredReport.req_document_guid,
+      document_due_date_type: requiredReport.req_document_due_date_type,
+      document_due_date_period_months: requiredReport.req_document_due_date_period_months,
+    };
```

Comment:
I feel like the backend should get the req_document_guid and go look up the due date fields itself, and not let the front end set them (we don't want the front end to change them do we)?